entrypoint/main.go

```diff
@@ -3,47 +3,86 @@
 package main
 
 import (
+	"bytes"
 	"compress/bzip2"
 	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
 	icingaSql "github.com/Icinga/go-libs/sql"
-	icingadb "github.com/Icinga/icingadb/config"
-	"github.com/go-ini/ini"
-	_ "github.com/go-sql-driver/mysql"
-	log "github.com/sirupsen/logrus"
+	icingadb "github.com/icinga/icingadb/pkg/config"
+	"go.uber.org/zap"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 	"regexp"
+	"sort"
+	"strconv"
 	"strings"
 	"syscall"
 )
 
 const config = "/etc/icingadb/config.ini"
 
 var myEnv = regexp.MustCompile(`(?s)\AICINGADB_(\w+?)_(\w+)=(.*)\z`)
+var isInt = regexp.MustCompile(`\A\d+\z`)
 var sqlComment = regexp.MustCompile(`(?m)^--.*`)
 var sqlStmtSep = regexp.MustCompile(`(?m);$`)
 
+var log = func() *zap.Logger {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+
+	return logger
+}()
+
 func main() {
-	log.SetOutput(os.Stdout)
-	log.SetLevel(log.TraceLevel)
-	log.Fatal(runDaemon())
+	log.Fatal("failed", zap.Error(runDaemon()))
 }
 
 func runDaemon() error {
-	log.Debug("translating env vars to .ini config")
+	log.Debug("translating env vars to YaML config")
 
-	cfg := ini.Empty()
+	cfg := map[string]map[string]interface{}{}
 	for _, env := range os.Environ() {
 		if match := myEnv.FindStringSubmatch(env); match != nil {
-			_, errNK := cfg.Section(strings.ToLower(match[1])).NewKey(strings.ToLower(match[2]), match[3])
-			if errNK != nil {
-				return errNK
+			section := strings.ToLower(match[1])
+
+			sectionCfg, ok := cfg[section]
+			if !ok {
+				sectionCfg = map[string]interface{}{}
+				cfg[section] = sectionCfg
+			}
+
+			rawValue := match[3]
+			var value interface{}
+
+			if isInt.MatchString(rawValue) {
```

Comment:
Why this additional check? Shouldn't this already be covered by ParseInt returning an error?